ports/rp2/boards/ARDUINO_NANO_RP2040_CONNECT/modules/lsm6dsox.py

```diff
@@ -0,0 +1,91 @@
+"""
+LSM6DSOX STMicro driver for MicroPython.
+Base on: LSM9DS1 driver and https://github.com/arduino-libraries/Arduino_LSM6DSOX
```

Comment:
Looks like that Arduino driver is LGPL.  It's not possible to put such code in this repo.

Reply:
Actually it's mostly based on this: https://github.com/hoihu/projects/blob/master/raspi-hat/lsm9ds1.py which is MIT..I only ported the init sequence from the Arduino driver, let me know if it still can't be added and maybe I can ask them to release another version with a compatible license.

Reply:
> mostly based on this: https://github.com/hoihu/projects/blob/master/raspi-hat/lsm9ds1.py

Haven't used it for a longer time, should hopefully still work.. no problem to use it anyway, I did the implementation straight from the datasheet at that time.

Reply:
@hoihu Yes it's still very much working :) I use it as well with the MIT license in your repo, and lsm6dsox is somewhat based on it (the lsm9ds1), but uses the init sequence from the Arduino library (with reference to both).

Reply:
rewrote it.